packages/xstate-graph/test/graph.test.ts

```diff
@@ -699,3 +603,99 @@ it('should provide previous state for serializeState()', () => {
       .map((plan) => plan.paths[0].steps.length)
   ).toEqual([0, 3]);
 });
+
+it.each([getShortestPlans, getSimplePlans])(
+  'initial state can be specified',
+  (getPlans) => {
+    const machine = createMachine({
+      initial: 'a',
+      states: {
+        a: {
+          on: { toB: 'b' }
+        },
+        b: {
+          on: { toC: 'c' }
+        },
+        c: {
+          on: { toA: 'a' }
+        }
+      }
+    });
+
+    const plans = getPlans(machine, {
+      initialState: machine.resolveState(State.from('b'))
```

Comment:
In general, I'm not super fond of specifying `initialState` like this. The problem itself is not that providing an initial state is bad per see but rather that we are able to use a somewhat arbitrary state as an initial state, one which might be invalid (not possible to achieve by transitioning from the true initial state) given the current config of the machine.

Maybe it's just something that the user has to be aware of. I just wonder if we can do any better? Nothing comes to my mind right now though because "proving" the initial state would be rather complex and not practical 🤔 

In the context of e2e tests though it should almost be impossible to do it. I assume that rarely one can "start" their model from an arbitrary point (joining paths is different since in there the provided `initialState` comes from the already generated paths that are starting in the true initial state.

Reply:
This is only meant to be "internal", specific to `@xstate/graph`. 

If you notice the test in `@xstate/test` (https://github.com/statelyai/xstate/pull/3446/files#diff-d7468424c510a8f0d4f4fcf0d2cbfe0ecda06e489f98982f501eb4275698a51eR259), you'll see that we're reaching the "custom initial state" by first getting the path to that desired state:

```js
const pathsToB = model.getShortestPathsTo((state) => state.matches('b'));
```

If we had `state.matches('someImpossibleState')`, the result will be an _empty array_, signaling that it is impossible to reach that state.

So in that sense, we already have safety for e2e tests.

Reply:
> This is only meant to be "internal", specific to @xstate/graph.

Note: it's still available as the option to `TestModel#getShortestPaths`. Perhaps my main concern here is the name - since it's not exactly the "initial" state in the same sense that we use the same name in XState itself. Maybe a `startState` or `traversalStartState` would be a better fit here (or at least a fit that would differentiate it from other places where we use "initial state" differently)

Reply:
That makes sense. I'd opt to call it `fromState`, as the goal is to find the paths from a specific state rather than from the true initial state.